web/src/search.ts

```diff
@@ -51,12 +51,22 @@ function narrow_or_search_for_term({on_narrow_search}: {on_narrow_search: OnNarr
     }
 
     assert(search_pill_widget !== null);
-    const search_query = search_pill.get_current_search_string_for_widget(search_pill_widget);
+    const search_query = [
+        search_pill.get_current_search_string_for_widget(search_pill_widget),
+        get_search_bar_text(),
+    ]
+        .filter((s) => s !== "")
+        .join(" ");
```

Comment:
This is kind of ugly -- is there a nicer way to write this?

Reply:
Is it intended to what it looks like it does, which is concatenate those strings with a `" "` between if they're both nonempty? Maybe something like `[x, y].join(" ").trim()` might be cleaner?

Reply:
Ah perfect, using trim sounds much better, thanks!